app/services/waste_notification.rb

```diff
@@ -42,8 +42,14 @@ def send_notifications
         p "Registration Found for Waste::LocationType: #{waste_pickup_time.waste_location_type.id}, #{registration_to_check.id} on #{waste_pickup_time.pickup_date}"
         p "Run at: #{date_time_to_run}"
 
+        # Look for existing Notification
+        existing_notifications = Delayed::Backend::ActiveRecord::Job
+                                   .where("handler LIKE '%WasteNotificationJob%'")
+                                   .where("handler LIKE '%args:\n- #{registration_to_check.id}\n- #{waste_pickup_time.id}%'")
+                                   .count
+
         # Send Notification
-        WasteNotificationJob.delay(run_at: date_time_to_run).perform(registration_to_check.id, waste_pickup_time.id)
+        WasteNotificationJob.delay(run_at: date_time_to_run).perform(registration_to_check.id, waste_pickup_time.id) if existing_notifications.zero?
```

Comment:
this line is too long, right? it is also hard to read what is going on with delay and perform until recognizing that there is an if statement.

i suggest something like

```
next if existing_notifications.any?
        
# Send Notification
WasteNotificationJob.delay(run_at: date_time_to_run).perform(registration_to_check.id, waste_pickup_time.id)
```

as we already have several "next if..." at the beginning of `waste_pickup_times.each do |waste_pickup_time|`